### PR TITLE
refactor(offline): Switch sync to use promise pooling and fine-grained progress % reporting

### PR DESCRIPTION
**Pull request by damassi**
### Description

In an effort to not future-spam our servers, this updates our offline sync code to use [promise-pool](https://github.com/supercharge/promise-pool), which comes with a default concurrency of 10. 

Also allows us to report much better download progress reporting to the user: 

https://user-images.githubusercontent.com/236943/211951129-586d78f4-3965-446c-bedf-31809fc78e08.mp4

cc @artsy/mobile-platform 

### PR Checklist

- [ ] I tested my changes on **iOS** and **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[docs]: /docs/README.md

### CODE REVIEW

File: src/app/components/AnimatedEllipsis.tsx [hunk 1]
```diff
@@ -78,6 +78,7 @@ const Dot = ({
   const defaultStyle: TextProps["style"] = {
     color: "black",
     fontSize: 26,
+    top: -1,
```
Comment (damassi) [inline src/app/components/AnimatedEllipsis.tsx:81]: The dots were hanging below the text in the progress status

**Review by joeyAghion — APPROVED** [review]

File: src/app/system/sync/syncManager.ts [hunk 2]
```diff
@@ -385,19 +405,28 @@ export function initSyncManager({
...
+          const QueryName = (error.req as RelayNetworkLayerRequest).id
+
+          return {
+            QueryName,
+            error,
+            ...rest,
+          }
+        }),
+        "QueryName"
+      )
```
Comment (damassi) [inline src/app/system/sync/syncManager.ts:424]: Fixes a bug where we were retrying too much. For a given sync query containing an array of items, multiple items could throw leading to multiple references of the same sync function being placed in the errors array. Make sure everything is uniq, and then retry _those_ 3 times.

File: src/app/system/sync/syncManager.ts [hunk 3]
```diff
@@ -385,19 +405,28 @@ export function initSyncManager({
 
     updateStatus("Retrying sync for errors")
 
-    const MAX_RETRY_ATTEMPTS = 3
+    const MAX_RETRY_ATTEMPTS = 2
```
Comment (damassi) [inline src/app/system/sync/syncManager.ts:408]: Now it's retry attempt 0, 1, 2. Was erroneously retrying 4 times before.

Comment (pvinis) [reply src/app/system/sync/syncManager.ts:410]: so its the initial try, plus 2 retries?

**Review by pvinis — APPROVED** [review]

**Comment by pvinis** [comment]
nice! assigning back, merge when you want.

**Pull request merged**